src/NuGet.Core/NuGet.Resolver/PackageResolver.cs

```diff
@@ -42,6 +43,30 @@ public IEnumerable<PackageIdentity> Resolve(PackageResolverContext context, Canc
                 }
             }
 
+            // validate existing package.config for any invalid dependency
+            var invalidsPackagesLst = new List<string>();
+            var installedPackages = context.PackagesConfig.Select(p => p.PackageIdentity);
```

Comment:
Add a ToArray, because it is going to get evaluated inside the loop multiple times.